File: backend/schemas/generation.py
from datetime import datetime

from pydantic import BaseModel


class GenerationRunOut(BaseModel):
    id: int
    chapter_id: int
    status: str
    final_output: str | None
    error_msg: str | None
    started_at: datetime | None
    completed_at: datetime | None
    created_at: datetime

    model_config = {"from_attributes": True}


class AgentMessageOut(BaseModel):
    id: int
    run_id: int
    sequence: int
    role: str
    content: str
    message_type: str
    created_at: datetime

    model_config = {"from_attributes": True}


class GenerationTriggerResponse(BaseModel):
    run_id: int

File: frontend/src/api/generation.ts
import client from './client'

export interface GenerationRun {
  id: number
  chapter_id: number
  status: string
  final_output: string | null
  error_msg: string | null
  started_at: string | null
  completed_at: string | null
  created_at: string
}

export interface AgentMessage {
  id: number
  run_id: number
  sequence: number
  role: string
  content: string
  message_type: string
  created_at: string
}

export const generationApi = {
  trigger: (chapterId: number) =>
    client.post<{ run_id: number }>(`/chapters/${chapterId}/generate`).then(r => r.data),
  stop: (runId: number) => client.post(`/runs/${runId}/stop`).then(r => r.data),
  listRuns: (chapterId: number) =>
    client.get<GenerationRun[]>(`/chapters/${chapterId}/runs`).then(r => r.data),
  getRun: (runId: number) => client.get<GenerationRun>(`/runs/${runId}`).then(r => r.data),
  getMessages: (runId: number) => client.get<AgentMessage[]>(`/runs/${runId}/messages`).then(r => r.data),
}
